libs/nim/verify-utils/verify_given_proof_constantine.nim

```diff
@@ -0,0 +1,134 @@
+  let a = ECP_ShortW_Aff[Fp[BN254_Snarks], G1].fromHex(a0,a1)
+  let b00 = BigInt[255].fromDecimal(proof["pi_b"][0][0].str).toHex()
+  let b01 = BigInt[255].fromDecimal(proof["pi_b"][0][1].str).toHex()
+  let b10 = BigInt[255].fromDecimal(proof["pi_b"][1][0].str).toHex()
+  let b11 = BigInt[255].fromDecimal(proof["pi_b"][1][1].str).toHex()
+  let b = ECP_ShortW_Aff[Fp2[BN254_Snarks], G2].fromHex(b00,b01,b10,b11)
+  let c0 = BigInt[255].fromDecimal(proof["pi_c"][0].str).toHex()
+  let c1 = BigInt[255].fromDecimal(proof["pi_c"][1].str).toHex()
+  let c = ECP_ShortW_Aff[Fp[BN254_Snarks], G1].fromHex(c0,c1)
+
+  Proof(a:a, b:b, c:c)
+
+proc createHeader*(pathCurrentHeader: string, updatePath: string, domain: string): Header =
+  let currentHeaderHashJSON = parseFile(pathCurrentHeader)
+  let updateJson = parseFile(updatePath)
+  let currentHeaderHash = hexToByteArray[32](currentHeaderHashJSON["attestedHeaderRoot"].str)
+  let newOptimisticHeader = hexToByteArray[32](updateJson["attestedHeaderRoot"].str)
+  let newFinalizedHeader = hexToByteArray[32](updateJson["finalizedHeaderRoot"].str)
+  let newExecutionStateRoot = hexToByteArray[32](updateJson["finalizedExecutionStateRoot"].str)
+  var slot = updateJson["attestedHeaderSlot"].getInt().toHex()
+  var currentSlot = hexToByteArray[8](slot)
+  var domain = hexToByteArray[32](domain)
+
+  var zerosSlotBuffer: array[24, byte]
+  for i in 0..23:
+    zerosSlotBuffer[i] = 0
+  let sha256ofHashes = hashHeaders(currentHeaderHash, newOptimisticHeader, newFinalizedHeader, newExecutionStateRoot, zerosSlotBuffer, currentSlot, domain)
+
+  headerFromSeq(@sha256ofHashes)
+
+proc makePairsAndVerify*(vk: VerificationKey,
+                         prf: Proof,
+                         header: Header): bool =
+
+  var preparedInputs:Input
+  var ic0Prj: ECP_ShortW_Prj[Fp[BN254_Snarks], G1]
+  var ic1Prj: ECP_ShortW_Prj[Fp[BN254_Snarks], G1]
+  var ic2Prj: ECP_ShortW_Prj[Fp[BN254_Snarks], G1]
+  fromAffine(ic0Prj, vk.ic[0])
+  fromAffine(ic1Prj, vk.ic[1])
+  fromAffine(ic2Prj, vk.ic[2])
+  scalarMul(ic1Prj,toBig(header.head))
+  scalarMul(ic2Prj,toBig(header.tail))
+  ic0Prj += ic1Prj
+  ic0Prj += ic2Prj
+  affine(preparedInputs.data,ic0Prj)
+
+  var aBPairing: Fp12[BN254_Snarks]
+  pairing_bn[BN254_Snarks](aBPairing,prf.a,prf.b)
+  var alphaBetaPairing: Fp12[BN254_Snarks]
+  pairing_bn[BN254_Snarks](alphaBetaPairing,vk.alpha,vk.beta)
+  var preparedInputsGammaPairing: Fp12[BN254_Snarks]
+  pairing_bn[BN254_Snarks](preparedInputsGammaPairing,preparedInputs.data,vk.gamma)
+  var proofCVkDeltaPairing: Fp12[BN254_Snarks]
+  pairing_bn[BN254_Snarks](proofCVkDeltaPairing,prf.c,vk.delta)
+
+  var sum:Fp12[BN254_Snarks]
+  prod(sum, alphaBetaPairing, preparedInputsGammaPairing)
+  prod(sum, sum, proofCVkDeltaPairing)
+
+  (sum == aBPairing).bool
+
+proc verifyProofByPathsConstantine*(pathToKey:string, pathToProof:string, pathToLastUpdate:string, pathToNewUpdate:string, domain:string): bool =
```

Comment:
I think `verifyProof` is a better name for this function. It is obvious we use path to files thanks to the parameter names.
What is more nim supports [overloading](https://nim-lang.org/docs/manual.html#overload-resolution) and if in the future we create other proc with different type of parameters we can use the same name. 

Also fix indentation